src/charm.py

```diff
@@ -409,13 +471,13 @@ def render_pgb_config(self, config: PgbConfig, reload_pgbouncer=False) -> None:
         self.peers.update_cfg(config)
 
         perm = 0o400
-        self.push_file(INI_PATH, config.render(), perm)
         for service in self._services:
             s_config = pgb.PgbConfig(config)
             s_config[PGB]["unix_socket_dir"] = service["dir"]
             s_config[PGB]["logfile"] = f"{service['dir']}/pgbouncer.log"
             s_config[PGB]["pidfile"] = f"{service['dir']}/pgbouncer.pid"
             self.push_file(service["ini_path"], s_config.render(), perm)
+        self.push_file(INI_PATH, config.render(), perm)
```

Comment:
Moved, because it is the only file that `_on_start()` checks for and sometimes the charm gets stuck if the service inis are not yet pushed to the container